### PR TITLE
Moose method modifier support

### PR DESCRIPTION
**Pull request by mephinet**
Thanks to this year's Perl advent calendar article http://www.perladvent.org/2014/2014-12-08.html I stumbled upon Perl::Metrics::Simple, and immediately gave it a try on several projects' codebases. While the output was very useful in general, it fell short for projects making use of Moose. The cause for that is that all methods that use Moose's method modifiers look like normal statements to P:M:S, which sums them all up and reports them as "{code not in named subroutines}". While this is technically correct, I think we can do better here, by teaching P:M:S how Moose's method modifiers look like and treating them as normal methods.
In short, when applying my proposed patch, for a method declare as `around 'foo' => sub {...}`, P:M:S sees the code as equivalent to `sub _around_foo {...}`, and reports it as such.
If you see any issues regarding style or quality of the patch, please let me know!

### CODE REVIEW

**Comment by matisse** [comment]
Very cool. I'll have a detailed look this weekend and go from there.
Thank you very much!

**Comment by matisse** [comment]
The chaining of the grep and map statements in Perl::Metrics::Simple::Analysis::File::_rewrite_moose_method_modifiers()
is very clever!

**Comment by matisse** [comment]
I'll see about releasing 0.18 tomorrow.

**Comment by mephinet** [comment]
Great, I'm glad you liked the patch, thanks for releasing it so quickly!
